src/storage/IndexBaseProcessor.inl

```diff
@@ -0,0 +1,188 @@
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "base/Base.h"
+#include "storage/IndexBaseProcessor.h"
+
+
+DECLARE_int32(bulk_number_per_index_creation);
+
+#define INDEX_CREATE_DONE                                              \
+do {                                                                   \
+    bool finished = false;                                             \
+    {                                                                  \
+        std::lock_guard<std::mutex> lg(this->lock_);                   \
+        if (thriftResult.code != cpp2::ErrorCode::SUCCEEDED) {         \
+        this->codes_.emplace_back(thriftResult);                       \
+        }                                                              \
+        this->callingNum_--;                                           \
+        if (this->callingNum_ == 0) {                                  \
+            this->result_.set_failed_codes(std::move(this->codes_));   \
+            finished = true;                                           \
+        }                                                              \
+    }                                                                  \
+    if (finished) {                                                    \
+        this->kvstore_->deleteSnapshot(spaceId_);                      \
+        this->onFinished();                                            \
+    }                                                                  \
+    return;                                                            \
+} while (false)                                                        \
+
+
+namespace nebula {
+namespace storage {
+
+template<typename RESP>
+void IndexBaseProcessor<RESP>::doIndexCreate(PartitionID partId) {
+    std::string key, val;
+    StatusOr<std::string> indexKey;
+    std::vector<kvstore::KV> data;
+
+    int32_t batchPutNum = 0;
+    cpp2::ResultCode thriftResult;
+    thriftResult.set_code(cpp2::ErrorCode::SUCCEEDED);
+    thriftResult.set_part_id(partId);
+
+    auto prefix = NebulaKeyUtils::partPrefix(partId);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = this->kvstore_->prefixSnapshot(spaceId_, partId, prefix, &iter);
+
+    if (ret != kvstore::ResultCode::SUCCEEDED || !iter) {
+        thriftResult.set_code(this->to(ret));
+        INDEX_CREATE_DONE;
+    }
+
+    while (iter->valid()) {
+        key = iter->key().str();
+        val = iter->val().str();
+        iter->next();
+        indexKey = (indexType_ == nebula::cpp2::IndexType::EDGE) ?
+                   assembleEdgeIndexKey(spaceId_, partId, key, std::move(val)) :
+                   assembleVertexIndexKey(spaceId_, partId, key, std::move(val));
+
+        if (!indexKey.ok()) {
+            continue;
+        }
+        data.emplace_back(std::move(indexKey.value()), std::move(key));
+        batchPutNum++;
```

Comment:
batchPutNum == data.size() ?

Reply:
Yep

Reply:
so batchPutNum is useless because can replace by data.size()